packages/core/src/auth/utils.ts

```diff
@@ -798,3 +798,13 @@ export function initializeCredentialsProviderManager() {
     manager.addProviderFactory(new SharedCredentialsProviderFactory())
     manager.addProviders(new Ec2CredentialsProvider(), new EcsCredentialsProvider(), new EnvVarsCredentialsProvider())
 }
+
+export async function getAuthType() {
+    let authType: CredentialSourceId | undefined = undefined
+    if (AuthUtil.instance.isEnterpriseSsoInUse() && AuthUtil.instance.isConnectionValid()) {
+        authType = 'iamIdentityCenter'
+    } else if (AuthUtil.instance.isBuilderIdInUse() && AuthUtil.instance.isConnectionValid()) {
+        authType = 'awsId'
+    }
+    return authType
+}
```

Comment:
This above function was previously in `transformApiHandler.ts`, just moving it here so it's in a shared utils file